test/register.js

```diff
@@ -78,3 +78,63 @@ test.cb('Utilize magic observer method on a component and observe state change',
 
   state.val = val
 })
+
+test('__regie_observer_removers__ exists on initialized component', t => {
+  const { $$register, state } = regie({})
+
+  const val = Math.random()
+
+  class Component {
+    constructor (state) {
+      this.props = state
+
+      this.created()
+    }
+
+    created () {
+      this.createdHooks()
+    }
+
+    ['observe val'] (newValue) {
+      t.is(newValue, val)
+      t.end()
+    }
+  }
+
+  $$register({ Component })
+
+  const comp = new Component(state)
+
+  t.true(Array.isArray(comp.__regie_observer_removers__))
+})
+
+test.cb('Remove magic observer when the component is disposed', t => {
+  const { $$register, state } = regie({})
+
+  class Component {
+    constructor (state) {
+      this.props = state
+
+      this.created()
+    }
+
+    created () {
+      this.createdHooks()
+    }
+
+    ['observe val'] () {
+      t.fail()
+    }
+
+    dispose () {}
+  }
+
+  $$register({ Component })
+
+  const comp = new Component(state)
+  comp.dispose()
+
+  state.val = Math.random()
+
```

Comment:
You might need a `t.pass()` here, no?

Reply:
Test was passing without any `.pass()` call. I guess it is because I did not call `.plan(x)` any number of assertions, but added `.pass()` call anyway. `.plan` is still unnecessary, because assertions are not in a loop or any subroutine.